app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -6,7 +6,7 @@ def initialize(records, find_key)
       # Save the columns associated with the find keys, so we can coerce the hash values during fetch
       if records.first
         model = records.first.class
-        @key_attribute_types = find_key.map { |k| model.type_for_attribute(k) }
+        @key_attribute_types = find_key.map { |k| model.type_for_attribute(k.to_s) }
```

Comment:
@cben hum this is funky, I am aware that type_for_attribute needs string. So that means the casting didn't work for quite some time. @agrare I wonder what we can break by enabling the casting :-)